StripePayments/StripePayments/Source/Helpers/STPCardValidator.swift

```diff
@@ -414,6 +414,43 @@ public class STPCardValidator: NSObject {
         return Set(brands)
     }
 
+    // This is a bit of a hack: We want to fetch BIN information for Card Brand Choice, but some
+    // of the BIN length information coming from the service is incorrect: We're receiving a maximum
+    // length, but we really should receive a min-max range.
+    // We don't want to pollute the main STPBINController cache with this bad data.
+    //
+    // We currently prevent cache pollution with an `isVariableLengthBINPrefix` check in
+    // `retrieveBinRanges()`, but we'll bypass that check when using the CBC BIN controller.
+    static let cbcBinController = STPBINController()
+
+    /// Returns available brands for the provided card details.
+    /// - Parameter card: The card details to validate.
+    /// - Parameter completion: Will be called with the set of available STPCardBrands or an error.
+    /// - seealso: https://stripe.com/docs/card-brand-choice
+    public class func possibleBrands(forCard cardParams: STPPaymentMethodCardParams,
+                                     completion: @escaping (Result<Set<STPCardBrand>, Error>) -> Void) {
+        guard let cardNumber = cardParams.number else {
+            completion(.success([]))
+            return
+        }
+        possibleBrands(forNumber: cardNumber, completion: completion)
+    }
+
+    class func possibleBrands(forNumber cardNumber: String,
+                              completion: @escaping (Result<Set<STPCardBrand>, Error>) -> Void) {
+        cbcBinController.retrieveBINRanges(forPrefix: cardNumber, recordErrorsAsSuccess: false, onlyFetchForVariableLengthBINs: false) { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+            case .success:
+                let binRanges = cbcBinController.binRanges(forNumber: cardNumber)
+                var brands = binRanges.map { $0.brand }
+                brands.removeAll { $0 == .unknown }
```

Comment:
This is maybe just stylistic so take it or leave it, but could also just chain this into a single let instead of mutating anything, i.e.

```suggestion
                let brands = binRanges.map { $0.brand }
                                      .filter { $0 != .unknown }
```